tokio/src/process/unix/mod.rs

```diff
@@ -115,8 +115,11 @@ impl fmt::Debug for Child {
     }
 }
 
-pub(crate) fn spawn_child(cmd: &mut std::process::Command) -> io::Result<SpawnedChild> {
-    let mut child = cmd.spawn()?;
+pub(crate) fn spawn_child_with(
+    cmd: &mut StdCommand,
+    with: impl Fn(&mut StdCommand) -> io::Result<StdChild>,
+) -> io::Result<SpawnedChild> {
+    let mut child = with(cmd)?;
```

Comment:
Might be worth reworking this slightly to avoid monomorphization bloat (basically the entire rest of the function body does not need to be re-monormorphized for every unique `with` that is passed in):

Something like:

```rust
pub(crate) fn spawn_child_with(
    cmd: &mut StdCommand,
    with: impl Fn(&mut StdCommand) -> io::Result<StdChild>,
) -> io::Result<SpawnedChild> {
    spawn_child(&mut with(cmd)?)
}

fn spawn_child(cmd: &mut StdCommand) -> io::Result<SpawnedChild> {
  // ...
}
```

Reply:
Here, you probably meant that the helper function takes an `std::process::Child` in since `with: impl Fn(&mut StdCommand) -> io::Result<StdChild>`, right? This is what I've used in the proposed update.